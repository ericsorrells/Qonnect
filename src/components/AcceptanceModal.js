// ========================================================================================
import React from 'react';
import Modal from 'react-modal';
// ========================================================================================
import ShowEventDisplay from '../components/ShowEventDisplay';
// ========================================================================================

class AcceptanceModal extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      acceptanceNote: ''
    }

    this.onNoteInput = this.onNoteInput.bind(this);
    this.onFormSubmit = this.onFormSubmit.bind(this);
  }

  onNoteInput(e) {
    event.preventDefault();
    this.setState({ acceptanceNote: e.target.value })
  }

  onFormSubmit(event){
    event.preventDefault();
    this.props.userAcceptEvent({
      acceptanceNote: this.state.acceptanceNote
    })
  }

  render(){
    return (
      <Modal
        isOpen={this.props.modalOpen}
        onRequestClose={this.props.closeModal}
        // style={customStyles}
        contentLabel="Example Modal"
      >
        <h2>Pleases Confirm You Wish To Accept This Event:</h2>
        <ShowEventDisplay event={this.props.event} />
        <form onSubmit={this.onFormSubmit}>
          <label>Acceptance Note:</label>
          <textarea
            onChange={this.onNoteInput}
            value={this.state.acceptanceNote}
          />
          <button type='submit'>Yes, I Accept!</button>
        </form>
      </Modal>
    )
  }
}

export default AcceptanceModal;