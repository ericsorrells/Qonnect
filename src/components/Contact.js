// ========================================================================================
import React from 'react';
// ========================================================================================

const Contact = () => {
  return(
    <h1>
      Coming Soon...
    </h1>
  )
}

export default Contact;