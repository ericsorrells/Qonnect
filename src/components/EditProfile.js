// ========================================================================================
import React from 'react';
import PropTypes from 'prop-types';
// ========================================================================================

class EditProfile extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      name:        '',
      birthday:    '',
      location:    '',
      joinedOn:    '',
      description: ''
    }
    
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleNameChange = this.handleNameChange.bind(this);
    this.handleLocationChange = this.handleLocationChange.bind(this);
    this.handleDescriptionChange = this.handleDescriptionChange.bind(this);
  }

  handleSubmit(e) {
    e.preventDefault();
    this.props.addProfile({
      name:        this.state.name,
      location:    this.state.location,
      description: this.state.description
    });
    this.setState({ name: '', location: '', description: '' })
  }

  handleNameChange(e) {
    e.preventDefault();
    this.setState({ name: e.target.value });
  }

  handleLocationChange(e) {
    e.preventDefault();
    this.setState({ location: e.target.value });
  }

  handleDescriptionChange(e) {
    e.preventDefault();
    this.setState({ description: e.target.value });
  }

  render() {
    return(
      <div>
        <form onSubmit={this.handleSubmit}>
          <input
            type        = "text"
            placeholder = "Add Your Name"
            value       = {this.state.name}
            onChange    = {this.handleNameChange}
          />
          <br/>
          <input
            type        = "text"
            placeholder = "Add Your Location"
            value       = {this.state.location}
            onChange    = {this.handleLocationChange}
          />
          <br/>
          <textarea
            type        = "text"
            placeholder = "Add Description"
            value       = {this.state.description}
            onChange    = {this.handleDescriptionChange}
          />
          <br/>
          <button>Add Profile</button>
        </form>
      </div>
    )
  }
}

EditProfile.propTypes = {
  profile: PropTypes.object
}

export default EditProfile;