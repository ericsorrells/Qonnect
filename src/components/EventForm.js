// ========================================================================================
import React from 'react';
import moment from 'moment';
import { SingleDatePicker } from 'react-dates';
// ========================================================================================

class EventForm extends React.Component {
  constructor(props) {
    super(props)

    const { event } = props;
    this.state = {
      eventName:   event ? event.eventName    : '',
      date:        event ? event.date         : '',
      time:        event ? event.time         : '',
      location:    event ? event.location     : '',
      category:    event ? event.category     : '',
      imageUrl:    event ? event.imageUrl     : '',
      description: event ? event.description  : '',
      focused: ''
    }

    this.onFormSubmit        = this.onFormSubmit.bind(this);
    this.onNameChange        = this.onNameChange.bind(this);
    this.onTimeChange        = this.onTimeChange.bind(this);
    this.onLocationChange    = this.onLocationChange.bind(this);
    this.onCategoryChange    = this.onCategoryChange.bind(this);
    this.onImageUrlChange    = this.onImageUrlChange.bind(this); 
    this.onDescriptionChange = this.onDescriptionChange.bind(this);
  }

  onFormSubmit(e) {
    e.preventDefault();
    const { firstName, lastName } = this.props.profile;

    this.props.onSubmit({
      userName:    `${firstName} ${lastName}`,
      uid:         this.props.auth.uid,
      eventName:   this.state.eventName,
      date:        moment(this.state.date).valueOf(),
      time:        this.state.time,
      location:    this.state.location,
      category:    this.state.category,
      imageUrl:    this.state.imageUrl,
      description: this.state.description
    })
  }

  onNameChange(e) {
    this.setState({ eventName: e.target.value })
  }

  onTimeChange(e) {
    console.log('TIME: ', e.target.value);
    this.setState({ time: e.target.value  })
  }

  onLocationChange(e) {
    this.setState({ location: e.target.value })
  }

  onCategoryChange(e){
    this.setState({ category: e.target.value })
  }

  onImageUrlChange(e) {
    this.setState({ imageUrl: e.target.value })
  }

  onDescriptionChange(e) {
    this.setState({ description: e.target.value })
  }

  onFocusChange({ focused }) {
    this.setState(() => ({ calendarFocused: focused }))
  }

  render() {
    return (
      <div className='form'>
        <form onSubmit={this.onFormSubmit} >
          <label> Event Name: </label>
          <input
            type='text'
            onChange={this.onNameChange}
            value={this.state.name}
          />

          <label> Event Date: </label>
          <SingleDatePicker
            date={this.state.date}
            onDateChange={ date => this.setState({ date: date }) }
            focused={this.state.focused}
            onFocusChange={({ focused }) => this.setState({ focused })}
            numberOfMonths={1}
            isOutsideRange={() => false} 
          />

          <label>Time:</label>
          <input type='time' onChange={this.onTimeChange}/>
          <label> Event Location: </label>
          <input
            type='text'
            onChange={this.onLocationChange}
            value={this.state.location}
          />

          <label>Category:</label>          
          <select onChange={this.onCategoryChange} value={this.state.category ? this.state.category : ''}>
            <option disabled selected value=''>Select A Category</option>
            <option value="Sports">Sports</option>
            <option value="Entertainment">Entertainment</option>
            <option value="Food">Food</option>
            <option value="Music">Music</option>
            <option value="Miscellaneous">Miscellaneous</option>
          </select>

          <label>Image URL:</label>
          <input
            type='text'
            onChange={this.onImageUrlChange}
            value={this.state.imageUrl}
          />

          <label> Description: </label>
          <textarea
            onChange={this.onDescriptionChange}
            value={this.state.description}
          />

          <button type='submit'>Submit</button>
        </form>
      </div>
    )
  }
}

export default EventForm;