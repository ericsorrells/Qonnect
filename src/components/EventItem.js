// ========================================================================================
import React     from 'react';
import PropTypes from 'prop-types';
import moment    from 'moment';
import { Link }  from 'react-router-dom';
// ========================================================================================
import Acceptances from './EventItem_Acceptances';
// ========================================================================================
import { formatTime, formatDate, formatText } from '../utils/utils';
// ========================================================================================

const EventItem = (event) => {
  const { id, uid, userName, imageUrl, eventName, location, description, time, date, history, interestedUsers, deleteEvent } = event;

  const deleteElement = () => {
    deleteEvent(id);                            // passed from showEvents props
  }

  const editElement = () => {
    history.push(`/edit-event/${id}`);          // passed from showEvents props
  }

  const showElement = () => {
    history.push(`/show-event/${id}`);
  }

  return (
    <li className='event-item'>
      <div className='event-item__left-container'>
        <div className='event-item__left'>
          <Image Link={Link} id={id} imageUrl={imageUrl} />
        </div>
        <div className='event-item__center'>
          <Link to={`/show-event/${id}`}>
            <Info value={eventName} type={'title'} />
          </Link>
          <Link to={`/profile/${uid}`}>
            <Info value={userName} type={'inviter'} />
          </Link>
          <div className='event-item__date_location'>
            {location && <SubInfo value={formatText(location, 20)} value2={null} type={'location'} icon={'map-marker'} />}
            {date && time && <SubInfo value={formatDate(date)} value2={`| ${formatTime(time)}`} type={'date'} icon={'calendar'} />  }
          </div>
          <Info value={formatText(description, 250)} type={'description'} />
        </div>
      </div>
      <div className='event-item__acceptances'>
        {interestedUsers && <Acceptances acceptances={Object.keys(interestedUsers).length} />}
      </div>
    </li>
  )
}

const Image = ({Link, id, imageUrl}) => {
 return(
    <Link to={`/show-event/${id}`}>
      {
        imageUrl ?
          <img src={imageUrl} className='event-item__image' /> :
          <img src='/images/default.png' className='event-item__image' />
      }
    </Link>
 )
}

const Info = ({ value, type }) => {
  return (
    <div className={`event-item__${type}`}>
      {value}
    </div>
  )
}

const SubInfo = ({ value, value2, type, icon }) => {
  return (
    <div className={`event-item__${type}`}>
      <div style={{ fontSize: '1em', color: '#6334e3' }}>
        <i className={`fas fa-${icon}`}></i>
      </div>
      <div>{value}   {value2}</div>
    </div>
  )
}

EventItem.propTypes = {
  id:          PropTypes.number,
  url:         PropTypes.string,
  name:        PropTypes.string,
  location:    PropTypes.string,
  description: PropTypes.string,
  inviter:     PropTypes.string,
  date:        PropTypes.string,
  history:     PropTypes.func,
  deleteEvent: PropTypes.func
};

export default EventItem;