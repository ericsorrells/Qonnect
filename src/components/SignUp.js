// ========================================================================================
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { auth } from '../firebase/firebaseIndex';
import { doSignUp, signIn } from '../actions/Auth';
import { updateProfile } from '../actions/Profile_Actions';
// ========================================================================================

const SignUp = ({ history, signIn, updateProfile }) => {
  return (
    <div className='profile_form sign-in-margins'>
      <h1>Sign Up</h1>
      <SignUpForm
        history={history}
        doSignUp={doSignUp}
        signIn={signIn}
        updateProfile={updateProfile}
      />
    </div>
  )
}

const INITIAL_STATE = {
  email: '',
  passwordOne: '',
  passwordTwo: '',
  error: null,
};

const byPropKey = (propertyName, value) => () => ({
  [propertyName]: value,
});

class SignUpForm extends Component {
  constructor(props) {
    super(props);
    this.state = { ...INITIAL_STATE };
    this.onSubmit = this.onSubmit.bind(this);
  }

  onSubmit(e) {
    e.preventDefault();
    const { email, passwordOne } = this.state;
    // TODO: WHY DOES THIS NOT WORK???
    // create signUp saga?
    // this.props.doSignUp(dispatch, email, passwordOne)

    return auth.doCreateUserWithEmailAndPasswor(email, passwordOne)
      .then((user) => {
        // this.props.signIn(user)    COMMENTED TO TEST FOR 'SIGN_IN' ACTION FIRINGS
        this.props.updateProfile({email: email})
        this.props.history.push('/update-profile')
      })
      .catch((error) => {
        console.error('ERROR SIGNING UP: ', error)
        this.setState({ error: error })
      });
  }

  render() {
    const { email, passwordOne, passwordTwo, error } = this.state;

    const isInvalid =
      passwordOne !== passwordTwo ||
      passwordOne === '' ||
      email === ''

    return (
      <form onSubmit={this.onSubmit}>
        <input
          value={email}
          onChange={event => this.setState(byPropKey('email', event.target.value))}
          type='text'
          placeholder='Email Address'
          className='form__element_border form__input'
        />
        <input
          value={passwordOne}
          onChange={event => this.setState(byPropKey('passwordOne', event.target.value))}
          type='password'
          placeholder='Password'
          className='form__element_border form__input'
        />
        <input
          value={passwordTwo}
          onChange={event => this.setState(byPropKey('passwordTwo', event.target.value))}
          type='password'
          placeholder='Confirm Password'
          className='form__element_border form__input'
        />
        <button disabled={isInvalid} type='submit'  className='button__purple'>
          Sign Up
        </button>
        {error && <p className='error'>{error.message}</p>}
      </form>
    );
  }
}

const SignUpLink = () =>
  <p>
    Don't have an account?
    {' '}
    <Link to={'/signup'}>Sign Up</Link>
  </p>

const mapDispatchToProps = (dispatch) => {
  return {
    signIn: (user) => dispatch(signIn(user.uid)),
    updateProfile: (profileUpdates) => dispatch(updateProfile(profileUpdates))
  }
}

SignUp.propTypes = {
  signIn:        PropTypes.func,
  updateProfile: PropTypes.func,
  history:       PropTypes.func
}

export { SignUpForm, SignUpLink };
export default connect(null, mapDispatchToProps)(SignUp);
