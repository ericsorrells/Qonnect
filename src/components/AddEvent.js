// ========================================================================================
import React from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
// ========================================================================================
import EventItems from './EventItems';
import EventForm from './EventForm';
import Spinner from './Spinner';
// ========================================================================================
import { getCurrentUser } from '../firebase/auth';
import { startGetProfile } from '../actions/Profile_Actions';
import { startCreateEvent } from '../actions/Events_Actions';
import { getEventUserIDs } from '../utils/utils';
// ========================================================================================

class AddEvent extends React.Component {
  constructor(props) {
    super(props)
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleSubmit(event) {
    this.props.startCreateEvent(event)
  }

  componentDidMount() {
    const { userId, events } = this.props;

    let eventIds = null;
    if (events) {
      eventIds = getEventUserIDs(events);
    }
    const hasEvents          = events.length === 0;
    const areUserEvents      = eventIds.includes(userId);

    if (userId && (hasEvents || !areUserEvents)) {
      this.props.startGetProfile(userId);
    }
  }

  render() {
    const component = this.props.loading.loading
      ? <Spinner />
      : <AddEventComponent {...this.props}
        events={this.props.events}
        history={this.props.history}
        deleteEvent={this.props.deleteEvent}
        onSubmit={this.handleSubmit}
        profile={this.props.profile}
        auth={this.props.auth}
      />
    return (
      <div>
        {component}
      </div>
    )
  }
}

const AddEventComponent = ({ events, history, deleteEvent, onSubmit, profile, auth }) => {
  return (
    <div className='add-event-container padding-left'>
      <aside>
        <EventForm
          event={null}
          onSubmit={onSubmit}
          profile={profile}
          auth={auth}
        />
      </aside>
      <section className='add-event__section'>
        <div className='add-event__inner-container'>
          {
            events &&
            <EventItems
              events={events}
              history={history}
              deleteEvent={deleteEvent}
            />
          }
        </div>
      </section>
    </div>
  )
}

AddEvent.propTypes = {
  events:      PropTypes.array,
  history:     PropTypes.func,
  deleteEvent: PropTypes.func,
  userId:      PropTypes.number,
  profile:     PropTypes.object,
  auth:        PropTypes.object,
  loading:     PropTypes.object
};

export { AddEventComponent };
export default AddEvent;