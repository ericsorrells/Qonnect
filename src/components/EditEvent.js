// ========================================================================================
import React from 'react';
// ========================================================================================
import EventForm from '../components/EventForm';
// ========================================================================================

// TODO: Can this be a stateless component???
class EditEvent extends React.Component {
  constructor(props) {
    super(props)
    this.onSubmit = this.onSubmit.bind(this);
  }

  onSubmit(event) {
    const { id, eventName, date, time, location, description } = event;
    // TODO: add support for persisting edited data to Firebase
    this.props.editEvent(id, { ...event })
    this.props.history.push(`/show-event/${id}`); 
  }

  render() {
    return (
      <div>
        <EventForm
          event={this.props.event}
          onSubmit={this.onSubmit}
          profile={this.props.profile}
          auth={this.props.auth}
        />
      </div>
    )
  }
}

export default EditEvent;