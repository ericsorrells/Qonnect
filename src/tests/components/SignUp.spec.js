// ========================================================================================
import React from 'react';
import { shallow, mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import mockStoreSetup from '../../utils/mockStoreHelpers';
import 'jest-enzyme';
// ========================================================================================
import { SignUpForm, SignUpLink } from '../../components/SignUp';
// ========================================================================================
import { basicEvent } from '../stubs/event';
import { basicProfile } from '../stubs/profile';
// ========================================================================================

// TODO TEST
describe('<SignUp/>', () => {
  const doSignUp      = jest.fn();
  const signIn        = jest.fn();
  const updateProfile = jest.fn();

  const component = mount(
    <SignUpForm
      history={jest.fn()}
      doSignUp={doSignUp}
      signIn={signIn}
      updateProfile={updateProfile}
    />
  )

  it('renders correctly', () => {
    expect(component).toMatchSnapshot();
  });

  describe('form', () => {
    it('calls doCreateUserWithEmailAndPassword()', () => {
        // TODO: fix this test
    });
  });
});