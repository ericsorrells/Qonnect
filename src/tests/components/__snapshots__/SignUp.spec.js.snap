// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUp/> renders correctly 1`] = `
<SignUpForm
  doSignUp={[MockFunction]}
  history={[MockFunction]}
  signIn={[MockFunction]}
  updateProfile={[MockFunction]}
>
  <form
    onSubmit={[Function]}
  >
    <input
      className="form__element_border form__input"
      onChange={[Function]}
      placeholder="Email Address"
      type="text"
      value=""
    />
    <input
      className="form__element_border form__input"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value=""
    />
    <input
      className="form__element_border form__input"
      onChange={[Function]}
      placeholder="Confirm Password"
      type="password"
      value=""
    />
    <button
      className="button__purple"
      disabled={true}
      type="submit"
    >
      Sign Up
    </button>
  </form>
</SignUpForm>
`;
